presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -171,6 +172,28 @@ default void renameColumn(ConnectorSession session, ConnectorTableHandle tableHa
     }
 
     /**
+     * Get the physical layout for a inserting into an existing table.
+     */
+    default Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        List<ConnectorTableLayout> layouts = getTableLayouts(session, tableHandle, new Constraint<>(TupleDomain.all(), map -> true), Optional.empty())
+                .stream()
+                .map(ConnectorTableLayoutResult::getTableLayout)
+                .filter(layout -> layout.getNodePartitioning().isPresent())
+                .collect(toList());
+
+        if (layouts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        if (layouts.size() > 1) {
+            throw new PrestoException(NOT_SUPPORTED, "Tables with multiple layouts can not be written");
```

Comment:
I agree, but I think @erichwang might have a connector that depends on this currently.  I'll check it in as is, since this is the existing behavior, but we should consider remove this.